script

```diff
@@ -784,12 +772,14 @@ only_send_error_notifications="0"
     # initialize vdisk_extensions_find_cmd as empty array.
     vdisk_extensions_find_cmd=()
 
+    local _addnl_ext
+    if [ "$inline_zstd_compress" -eq 1 ]; then
+      _addnl_ext=".zst"
+    fi
+
```

Comment:
I don't think this will work. The find command seems to be malformed, and the zst extension is getting added multiple times. instead, I would do this:


```
    # initialize vdisk_extensions_find_cmd as empty array.
    vdisk_extensions_find_cmd=()

    # find each vdisk extension and use it to build a find command.
    for extension in "${vdisk_extensions[@]}"
    do

      # check to see if find command is empty.
      if [ ${#vdisk_extensions_find_cmd[@]} -eq 0 ]; then

        # build initial find command.
        vdisk_extensions_find_cmd=(find)
        vdisk_extensions_find_cmd+=("$search_directory")
        vdisk_extensions_find_cmd+=(-type f)
        vdisk_extensions_find_cmd+=(\()
        vdisk_extensions_find_cmd+=(-name '*.'"$extension")

      else

        # add additional extensions to find command.
        vdisk_extensions_find_cmd+=(-or -name '*.'"$extension")
      fi
    done

    if [ "$inline_zstd_compress" -eq 1 ]; then
      vdisk_extensions_find_cmd+=(-or -name '*.zst')
    fi

    # put closing parenthesis on find command.
    vdisk_extensions_find_cmd+=(\))

```

Reply:
That doesn't look equivalent.  I want to filter for things like `-name '*.img.zst' -or -name '*.raw.zst'` instead of `-name '*.img' -or -name '*.raw' -or -name '*.zst'`.

But, I just committed a change that should work better?